tests/ETHTornado.test.ts

```diff
@@ -2,57 +2,66 @@ import { assert, expect } from "chai";
 import { ETHTornado__factory, Verifier__factory, ETHTornado } from "../types/";
 
 import { ethers } from "hardhat";
-import { ContractFactory, BigNumber, BigNumberish } from "ethers";
+import { Contract, ContractFactory, BigNumber, BigNumberish } from "ethers";
 // @ts-ignore
-import { createCode, generateABI } from "circomlib/src/poseidon_gencontract";
+import { poseidonContract, buildPoseidon } from "circomlibjs";
 // @ts-ignore
-import poseidon from "circomlib/src/poseidon";
 import { MerkleTree, Hasher } from "../src/merkleTree";
 // @ts-ignore
 import { groth16 } from "snarkjs";
 import path from "path";
 
+import { readFileSync } from "fs";
+
 const ETH_AMOUNT = ethers.utils.parseEther("1");
 const HEIGHT = 20;
 
-function poseidonHash(inputs: BigNumberish[]): string {
+function poseidonHash(poseidon: any, inputs: BigNumberish[]): string {
     const hash = poseidon(inputs.map((x) => BigNumber.from(x).toBigInt()));
-    const bytes32 = ethers.utils.hexZeroPad(
-        BigNumber.from(hash).toHexString(),
-        32
-    );
-    return bytes32;
+    const hashStr = poseidon.F.toString(hash);
+    const hashHex = BigNumber.from(hashStr).toHexString();
+
+    return hashHex;
```

Comment:
```suggestion
    // pad zero to make it 32 bytes, so that the output can be taken as a bytes32 contract argument
    const bytes32 = ethers.utils.hexZeroPad(
        BigNumber.from(hash).toHexString(),
        32
    );
    return bytes32;
```